### PR TITLE
*: update pd client and client-go

### PR DESCRIPTION
**Pull request by nolouch**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/48598

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

image

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by CabinfeverB — APPROVED** [review]

**Review by HuSharp — APPROVED** [review]

**Review by hawkingrei — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

File: pkg/domain/runaway.go [hunk 1]
```diff
@@ -399,7 +399,7 @@ func (do *Domain) initResourceGroupsController(ctx context.Context, pdClient pd.
 		return nil
 	}
 
-	control, err := rmclient.NewResourceGroupController(ctx, uniqueID, pdClient, nil, rmclient.WithMaxWaitDuration(resourcegroup.MaxWaitDuration))
+	control, err := rmclient.NewResourceGroupController(ctx, uniqueID, pdClient, nil)
```
Comment (CabinfeverB) [inline pkg/domain/runaway.go:402]: We replace `LoadGlobalConfig` with `Get`, so need to impl `Get` in mock pd client

**Comment by CabinfeverB** [comment]
`pdClient` in /pkg/store/mockstore/unistore/pd.go

**Comment by CabinfeverB** [comment]
sry, there are other circumstances. `TestResolvedLargeTxnLocks` uses mock pd client in client-go not in tidb

File: pkg/domain/runaway.go (hunk 1 above)
Comment (nolouch) [reply pkg/domain/runaway.go:402]: updated to the latest pd client in release-7.5

**Pull request merged**